pkg/router/template/template_helper.go

```diff
@@ -409,8 +409,8 @@ var helperFunctions = template.FuncMap{
 	"getPrimaryAliasKey":          getPrimaryAliasKey,          //returns the key of the primary alias for a group of aliases
 
 	"generateHAProxyMap":           generateHAProxyMap,           //generates a haproxy map content
-	"validateHAProxyWhiteList":     validateHAProxyWhiteList,     //validates a haproxy whitelist (acl) content
-	"generateHAProxyWhiteListFile": generateHAProxyWhiteListFile, //generates a haproxy whitelist file for use in an acl
+	"validateHAProxyAllowlist":     validateHAProxyAllowlist,     //validates a haproxy allowlist (acl) content
```

Comment:
Are you concerned about breaking custom templates at all? I don't think we officially support any form of custom template, so don't have to maintain compatibility?

If it actually is a concern, you could leave "validateHAProxyWhiteList" as an entry in this `template.FuncMap` to maintain support for custom template. I think its okay to break custom templates though.

Reply:
> Are you concerned about breaking custom templates at all?

A little bit.  

> I don't think we officially support any form of custom template, so don't have to maintain compatibility?

We absolutely do not, and never did, _support_ custom templates.  That said, it's entirely possible to run an unsupported custom router deployment on OCP, and then of course there's OKD.  Support concerns aside, it is unkind to break custom config capriciously, be it customers' or community members'.  For what it's worth, when I carelessly broke the dynamic config manager, which isn't enabled in OCP 4, in #51, someone noticed (and then graciously fixed it for us; see #107).  However, I do not believe we are being capricious or careless in this case.  

> If it actually is a concern, you could leave "validateHAProxyWhiteList" as an entry in this `template.FuncMap` to maintain support for custom template. I think its okay to break custom templates though.

Yeah, it would be trivial to maintain backwards compatibility.  It would be slightly more effort to add a "this helper is deprecated" log message.  

Normally I would push back against any kind of breaking change.  However, the goal here is _specifically_ to remove the old names, we _will_ document the change in a release note, and it is trivial to update one's custom template once one realizes it is necessary to do so (either from seeing the release note or from testing the custom template in the new .y release before deploying one's unsupported custom template to a production environment).  

One more thing to consider is that we cannot really force a migration for unsupported config.  At most we could start logging an error message in one release and then remove the helper entirely in a later release.  However, the fact that the thing we are talking about here is a custom router deployment means that it isn't being managed by a version of cluster-ingress-operator that we maintain, so our operator cannot necessarily detect the problematic config in order to block an upgrade.  

In light of all that, does it seem reasonable just to remove the helpers?

Reply:
Yea that sounds convincing enough, I just wanted to be sure. To reiterate:
1. This change is different, we are specifically trying to change the function name here
2. Custom templates are unsupported
3. We'll do a release note
4. We don't have a reasonable way to force a migration for these situations anyways